code/obj/machinery/manufacturer.dm

```diff
@@ -648,6 +674,81 @@
 					src.pulse(twire)
 				src.build_icon()
 
+			if (href_list["card"])
+				if (src.scan) src.scan = null
+				else
+					var/obj/item/I = usr.equipped()
+					src.scan_card(I)
+
+			if (href_list["purchase"])
+				var/obj/machinery/ore_cloud_storage_container/storage = locate(href_list["storage"])
+				var/ore = href_list["ore"]
+				var/datum/ore_cloud_data/OCD = storage.ores[ore]
+				var/price = OCD.price
+				var/taxes = round(max(rockbox_globals.rockbox_client_fee_min,abs(price*rockbox_globals.rockbox_client_fee_pct/100)),0.01) //transaction taxes for the station budget
+
+				if(!storage)
+					return
+
+				if(storage.broken)
+					return
+
+				if(!scan)
+					src.temp = {"You have to scan a card in first.<BR>"}
+					src.updateUsrDialog()
+					return
+				else
+					src.temp = null
+				if (src.scan.registered in FrozenAccounts)
+					boutput(usr, "<span class='alert'>Your account cannot currently be liquidated due to active borrows.</span>")
+					return
+				var/datum/data/record/account = null
+				account = FindBankAccountByName(src.scan.registered)
+				if (account)
+					var/quantity = 1
+					quantity = input("How many units do you want to purchase?", "Ore Purchase", null, null) as num
+
+					////////////
+
+					if(OCD.amount >= quantity)
+						var/subtotal = round(price * quantity)
+						var/sum_taxes = round(taxes * quantity)
+						var/rockbox_fees = (!rockbox_globals.rockbox_premium_purchased ? ROCKBOX_STANDARD_FEE : 0) * quantity
+						var/total = subtotal + sum_taxes + rockbox_fees
+						if(account.fields["current_money"] >= total)
+							account.fields["current_money"] -= total
+							storage.eject_ores(ore, get_output_location(), quantity, transmit=1, user=usr)
```

Comment:
Might be nice if (there was the option to have) ores bought remotely be auto-processed, since I don't think you can shove raw materials into a manufacturer

Reply:
> Additionally, I'm not sure how well this handles rematerialed ores (e.g.: midas touch turning a mauxite ore into 'gold' material)

I am going to scream

Reply:
I also think that would be nice, but that would involve having to navigate the heck that is materials processing 😅 

Maybe later

Reply:
fixed the gold thing in pending commit. Let's see how this works without a reclaimer first